### ISSUE TITLE
UUID Support

### ISSUE THREAD

**Issue by waggy55** [smallrye/smallrye-open-api#193]
At present, it seems the UUID type is not recognised. For projects that define the API in yaml, this is not a problem, because the type can be defined in the yaml file. However, for projects where the API is created by parsing the code, the UUID type is represented as:
```
{
   "leastSigBits": 0,
    "mostSigBits": 0,
    "leastSignificantBits": 0,
    "mostSignificantBits": 0
}
```
This is quite unusable (at least not quick to use). For microservices environments, the API often contains UUIDs, so support for this would be much appreciated.

Many thanks,
Dan

**Comment by MikeEdgar** [comment]
@waggy55 - are you using Quarkus?

**Comment by waggy55** [comment]
@MikeEdgar yes I am

**Comment by MikeEdgar** [comment]
Thank you. I'm assuming you're expecting to see the UUID type represented as a string?

We are going to look at other ways to account for the fact that the class index being passed in contains JDK classes that the unit tests assume are not there.

**Comment by waggy55** [comment]
Yes, to see the UUID as a string would be great! Thank you for adding it to "To do".

**Comment by MikeEdgar** [comment]
@EricWittmann , @msavy - can either of you think of a reason that `java.util.UUID` should not be added to the `TypeUtil` `TYPE_MAP` with a value of `STRING_FORMAT`? I think this is a pretty common type and it is supported by JAX-RS out of the box via its static `fromString` method for parsing.

**Comment by EricWittmann** [comment]
I can't think of a reason.  The only thing I would say is that it might be nice to map UUID to a more interesting schema type that perhaps validates the format of a typical Java UUID.  In other words, rather than just map it to string, maybe we can do better and generate a re-usable type that is a string with a regex pattern or some such.

Not a big deal for me - just a thought.  I'm more than happy mapping it to string.

**Comment by MikeEdgar** [comment]
>  In other words, rather than just map it to string, maybe we can do better and generate a re-usable type that is a string with a regex pattern or some such.

I really like this idea. This has been implemented in #195. I think there may be a number of classes that fall into this category and adjusting the approach for eclipse/microprofile-open-api#364 may be a good idea as well.
